design/ceph-deprecate-old-osd-types.md

```diff
@@ -0,0 +1,433 @@
+for such environments.
+
+To enable this feature, the Rook-Ceph operator will observe the environment variable
+`TEST_ENVIRONMENT_FILE_DEVICE_SIZE_GiB`. If this variable is set with a positive, nonzero value, the
+operator will not search hosts for devices. It will ignore `useAllDevices`, `deviceFilter`, and any
+`devices` directives in the Ceph cluster config and instead create a file in the `dataDirHostPath`
+with a size determined by the above environment variable measured in Gibibytes. This file will be
+provisioned as a Bluestore OSD.
+
+This option is similar to Rook's current usage of directory-based OSDs, but it is critically
+different in that it is explicitly reserved for test environment usage and is not made to be as
+configurable or flexible as the supported disk usage. This inflexibility is important for keeping
+the new OSD implementation as clean, extensible, and maintainable as possible for production
+features.
+
+#### Operator knowledge
+Currently, the config updates are limited because the Rook operator has limited knowledge of the
+hardware being configured on hosts. The updated config plan calls for a Ceph config file that is
+minimal and standard for all daemons; daemons are individually configured using command line flags.
+In order to properly set command line flags for object storage daemons, the operator must have
+knowledge of the disks which an OSD will use. The operator must configure, for example, an OSD to
+use one disk as the main storage, another disk as wal storage, and another disk as db storage.
+Mechanisms are not currently in place to report the needed disk information to the operator.
+
+
+### Proposed Rook operator design
+The existing OSD operator design uses a two-stage process. The first stage is preparation of OSDs on
+a per-host basis, and the second stage is running individual OSDs. The design proposed here
+separates OSD tasks into more stages. The stages are numbered from the perspective of the
+**Rook-managed mode**, and not all stages will apply to the **Ceph-managed mode**.
+
+A Rook OSD prepare binary currently handles the equivalent of stages 1 through 4, and in early
+implementations of the new OSD design, a version of the current binary could be used for stages 1
+through 3 to achieve the minimum viable product more quickly. Splitting the stages can be done
+afterwards as makes sense.
+
+#### Stage 1: Host device inspection (Rook-managed mode only)
+The first stage is an inspection stage that is currently done as part of the existing preparation
+stage. Because all of the host's devices must be inspected, this stage likely needs access to the
+entirety of the host's `/dev` directory. It may also need access to large sections of other
+directories like `/run/udev`.
+
+Similar to the current Rook OSD prepare binary, this proposal calls for a Rook inspection binary
+which will examine the disk hardware, rejecting disks that are already in use by applications other
```

Comment:
We need to highlight that as per https://github.com/rook/rook/issues/2606 the device detection will use `ceph-volume inventory` command instead of the current detection with `lsblk`.